sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -317,15 +358,18 @@ class StreamExecution(
     // TODO: Move this to IncrementalExecution.
 
     // Request unprocessed data from all sources.
-    val newData = availableOffsets.flatMap {
-      case (source, available)
+    val newData = timeIt(GET_BATCH_LATENCY) {
```

Comment:
Note that the time interval being measured here will have different semantics for different sources, depending on how much computation occurs inside the source's `getBatch` method vs. lazily when the data is read from the resulting Dataframe.